app/models/miq_db_config.rb

```diff
@@ -175,69 +175,50 @@ def method_missing(m, *args)
     end
   end
 
-  def opt_file_for_conn_test(from_save, &blk)
-    raise "must pass block" unless block_given?
-    opts = self.class.raw_config['production']
+  def with_temporary_connection(config_hash = nil)
+    config_hash ||= self.class.raw_config['production']
 
-    begin
-      FileUtils.mkdir_p(File.dirname(DB_FILE))
-      File.delete(DB_FILE) if File.exist?(DB_FILE)
-      opts[:from_save] = from_save
-      File.open(DB_FILE, "wb") {|f|f.write(Base64.encode64(Marshal.dump(opts)))}
-      res = yield
-    ensure
-      File.delete(DB_FILE) if File.exist?(DB_FILE)
-    end
-    res
-  end
-
-  def delete_io_files
-    [IO_DOLLAR_STDOUT, IO_DOLLAR_STDERR].each {|f| File.delete(f) if File.exist?(f)}
+    before = ActiveRecord::Tasks::DatabaseTasks.database_configuration
+    ActiveRecord::Tasks::DatabaseTasks.database_configuration = config_hash
+    conn = ActiveRecord::Base.establish_connection(config_hash).connection
```

Comment:
Doesn't this establish a connection in the same pool as the rest of the models?  Might be ok if this runs in an external task, but if you want to use models afterward in the same process, I think this will be a problem.  In other places where we need a fake connection, we create a new dummy class that derives from ActiveRecord::Base, and then establish the connection pool on that.

@tenderlove @matthewd Please correct me if I'm wrong with respect to Rails 4.